journal.py

```diff
@@ -92,45 +103,94 @@ def home(request):
     return {'entries': entries}
 
 
-@view_config(route_name='add', renderer='templates/add.jinja2')
-def add(request):
+def do_add(request, redirect):
     if request.method == 'POST':
         title = request.params.get('title')
         text = request.params.get('text')
         if not (title == "" or text == ""):
             Entry.write(title=title, text=text)
-            return HTTPFound(request.route_url('home'))
+            if redirect:
+                return HTTPFound(request.route_url('home'))
+            else:
+                new_entry = Entry.last()
+                return Response(body=json.dumps({
+                                'title': title,
+                                'text': text,
+                                'id': new_entry.id,
+                                'time': new_entry.created.strftime('%b. %d, %Y')}),
+                                content_type=b'application/json')
         else:
             return {'title': title, 'text': text}
     else:
         return {'title': '', 'text': ''}
 
 
+@view_config(route_name='add', renderer='templates/add.jinja2')
+def add(request):
+    return do_add(request, True)
+
+
+@view_config(route_name='ajax_add', renderer='templates/ajax_add.jinja2')
```

Comment:
You could totally make this use the same route by adding the `xhr=True` predicate to your `view_config`.  Also, an ajax response should probably not be using a jinja template renderer.